File: Delete node in Doubly Linked List.cpp
class Solution
{
    public:
    Node* deleteNode(Node *head_ref, int x)
    {
      
      if(x==1)
      {
          head_ref=head_ref->next;
          return head_ref;
      }
      
      int count=0;
      
      Node*temp=head_ref;
      
      while(count<x && temp)
      {
          count++;
          if(count==x)
          {
              temp->prev->next=temp->next;
              return head_ref;
              
          }
              
          else
          temp=temp->next;
      }
    }
};

File: JAVA/Delete node in Doubly Linked List.java
class Solution {
    public Node deleteNode(Node head, int x) {
        if(x==1){
          return head.next;
        }
        Node temp = head;
        for(int i = 1;i<x-1;i++){
            temp = temp.next;
        }
         temp.next = temp.next.next;
         if(temp.next!=null){
             temp.next.prev = temp;
         }
        return head;
    }
}
